### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by ikushlianski**
- [x] 1 - File serverless.yml contains configuration for catalogBatchProcess function
- [x] 2 - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [x] 2 - File serverless.yml contains policies to allow lambda catalogBatchProcess function to interact with SNS and SQS
- [x] 3 - File serverless.yml contains configuration for SQS catalogItemsQueue
- [x] 4 - File serverless.yml contains configuration for SNS Topic createProductTopic and email subscription
---
- [x] +1 - catalogBatchProcess lambda is covered by unit tests
- [x] +1 - set a Filter Policy for SNS createProductTopic in serverless.yml (Create an additional email subscription and distribute messages to different emails depending on the filter for any product attribute)

### CODE REVIEW

File: packages/import-service/handlers/catalog-batch-process.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { pgDatabaseService } from '../services/pg-database.service';
+import { getErrorCode } from '../../shared/src/errors';
+import AWS from 'aws-sdk';
+
+const sns = new AWS.SNS();
+
+export const catalogBatchProcess = async (event) => {
```
Comment (MaxPlav) [inline packages/import-service/handlers/catalog-batch-process.js]: according to the task, 'catalogBatchProcess' should be a part of 'product-service' service

File: packages/import-service/serverless.yml [hunk 2]
```diff
@@ -18,11 +18,60 @@ provider:
...
+      Resource:
+        Ref: createProductTopic
+
+resources:
+  Resources:
+    catalogItemsQueue:
+      Type: "AWS::SQS::Queue"
+      Properties:
+        QueueName: catalogItemsQueue
+    createProductTopic:
```
Comment (MaxPlav) [inline packages/import-service/serverless.yml]: SNS resource should be a part of 'product-service'

File: packages/import-service/handlers/catalog-batch-process.js (hunk 1 above)
Comment (ikushlianski) [reply packages/import-service/handlers/catalog-batch-process.js]: I re-read the instructions https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task6-sqs-sns/task.md and don't see where this requirement is written

Comment (MaxPlav) [reply packages/import-service/handlers/catalog-batch-process.js]: it was mentioned in the video and the diagram of the 6th task https://www.youtube.com/watch?v=Fu9txvDvsDY&t=5333s
Screenshot 2021-05-24 at 21 45 34
"Update lambda" on the diagram is "catalogBatchProcess"

File: packages/import-service/serverless.yml (hunk 2 above)
Comment (ikushlianski) [reply packages/import-service/serverless.yml]: fixed

File: packages/import-service/handlers/catalog-batch-process.js (hunk 1 above)
Comment (ikushlianski) [reply packages/import-service/handlers/catalog-batch-process.js]: Thanks for noticing

**Pull request merged**